torch/csrc/distributed/rpc/python_rpc_handler.cpp

```diff
@@ -26,6 +26,21 @@ PythonRpcHandler::PythonRpcHandler() {
   pySerialize_ = getFunction(module, "serialize");
 }
 
+// In default, PythonRpcHandler will call dec_ref() to clean up python objects.
+// It is found that PythonRpcHandler will dec_ref() null python objects and
+// crashed when program exits in Python 3.5 only (The theory is: the python
+// objects are cleaned up before destructing PythonRpcHandler when program
+// exits in Python 3.5).
+// To avoid PythonRpcHandler destructor to call dec_ref(), explicitly assign
+// the py::handle to be none. This will not have memory leak, as when destruting
+// PythonRpcHandler singleton when program exits, all memories will be cleaned
+// up by OS.
+PythonRpcHandler::~PythonRpcHandler() {
+  pyRunFunction_ = py::none();
```

Comment:
Do we need GIL here?

Reply:
no, I tired to use GIL, it will throw error

Reply:
what error did you see? It sounds suspicious that adding synchronization triggers errors.

Reply:
my understanding is: at this point, python interpreter may be gone?

Reply:
some python system objects and states are not there anymore at this point

Reply:
I see